docs/resources/gke_cluster.md

```diff
@@ -0,0 +1,458 @@
+      }
+      node_pools {
+        name         = "np"
+        node_version = "1.26"
+        size         = 3
+        machine_config {
+          machine_type   = "e2-standard-4"
+          image_type     = "COS_CONTAINERD"
+          boot_disk_type = "pd-standard"
+          boot_disk_size = 100
+        }
+        management {
+          auto_upgrade = "true"
+        }
+        upgrade_settings {
+          strategy = "SURGE"
+          surge_settings {
+            max_surge       = 0
+            max_unavailable = 1
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Argument Reference
+
+### Required
+
+- `metadata` - (Block List, Max: 1) Contains data that helps uniquely identify the resource. (See [below for nested schema](#nestedblock--metadata))
+- `spec` - (Block List, Max: 1) Defines the characteristics for the resource. This is the desired state for the resource. (See [below for nested schema](#nestedblock--spec))
+
+### Optional
+
+- `timeouts` - (Block) Sets the duration of time the create, delete, and update functions are allowed to run. If the function takes longer than this, it is assumed the function has failed. The default is 10 minutes. (See [below for nested schema](#nestedblock--timeouts))
+
+
+### Read-Only
+
+- `id` (String) The ID of this resource.
+
+<a id="nestedblock--metadata"></a>
+### Nested Schema for `metadata`
+
+
+***Required***
+
+- `name` - (String) The name of the AKS Cluster. This must be unique in your organization.
+- `project` - (String) The name of the Rafay project the cluster will be created in.
+
+***Optional***
+
+- `annotations` (Map of String) annotations of the resource
+- `description` (String) description of the resource
+- `labels` (Map of String) labels of the resource
+
+<a id="nestedblock--spec"></a>
+### Nested Schema for `spec`
+
+
+***Required***
+
+- `cloud_credentials` - (String) The name of the cloud credentials used to create and manage the cluster.
+- `config` - (Block List, Min: 1) The AKS specific cluster configuration. (See [below for nested schema](#nestedblock--spec--config))
+- `type` - (String) The GKE Cluster type. The supported value is `gke`.
+- `blueprint` (Block List, Max: 1) The blueprint to be used for this cluster. Use this for GKE cluster type (see [below for nested schema](#nestedblock--spec--blueprint))
+
+***Optional***
+
+- `proxy` (Block List, Max: 1) The proxy to be used for this cluster. Use this for GKE cluster type (see [below for nested schema](#nestedblock--spec--proxy))
+- `sharing` (Block List, Max: 1) Sharing spec to be used for sharing the cluster with projects (see [below for nested schema](#nestedblock--spec--sharing))
+
+<a id="nestedblock--spec--blueprint"></a>
+### Nested Schema for `spec.blueprint`
+
+***Required***
+
+- `name` (String)
+- `version` (String)
+
+<a id="nestedblock--spec--proxy"></a>
+### Nested Schema for `spec.proxy`
+
+Optional:
+
```

Comment:
We cannot have everything optional in proxy block. Need to add help text for all proxy fields